mempool/cat/pool_test.go

```diff
@@ -782,3 +782,94 @@ func TestTxPool_BroadcastQueue(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestTxPool_IsRejectedTx(t *testing.T) {
+	t.Run("CheckTx rejection", func(t *testing.T) {
+		txmp := setup(t, 100)
+
+		// Create a transaction that will be rejected by CheckTx (invalid format)
+		rejectedTx := []byte("invalid-tx-format")
+		txKey := types.Tx(rejectedTx).Key()
+
+		// The transaction should not be rejected initially
+		require.False(t, txmp.IsRejectedTx(txKey))
+
+		// Try to add the transaction - it should be rejected by the application
+		err := txmp.CheckTx(rejectedTx, nil, mempool.TxInfo{})
+		require.Error(t, err) // CheckTx method itself doesn't return error for app rejection
```

Comment:
I don't understand this comment. In this case `CheckTx` is returning an error because the line is asserting that an error was returned.

```suggestion
		require.Error(t, err)
```

Reply:
Yeah app CheckTx errors are returned https://github.com/celestiaorg/celestia-core/blob/6a5acc9dac0bfec405b954bb54cd97f2bc662796/mempool/priority/mempool.go#L225